src/snowflake/cli/api/entities/utils.py

```diff
@@ -41,40 +44,15 @@
 from snowflake.connector.cursor import SnowflakeCursor
 
 
-def generic_sql_error_handler(
-    err: ProgrammingError, role: Optional[str] = None, warehouse: Optional[str] = None
-) -> NoReturn:
+def generic_sql_error_handler(err: ProgrammingError) -> NoReturn:
     # Potential refactor: If moving away from Python 3.8 and 3.9 to >= 3.10, use match ... case
-    if err.errno == DOES_NOT_EXIST_OR_CANNOT_BE_PERFORMED:
-        raise ProgrammingError(
-            msg=dedent(
-                f"""\
-                Received error message '{err.msg}' while executing SQL statement.
-                '{role}' may not have access to warehouse '{warehouse}'.
-                Please grant usage privilege on warehouse to this role.
-                """
-            ),
-            errno=err.errno,
-        )
+    if (
+        err.errno == DOES_NOT_EXIST_OR_CANNOT_BE_PERFORMED
+        or "does not exist or not authorized" in err.msg
```

Comment:
Why `errno` is not enough?

Reply:
I did not write the original code and the person who did is on leave for a couple of weeks, so it's hard to say, but there may be some errors raised where the errno is not set or does not match what is expected. However, the two cases are similar enough that we decided to refactor the two branches into one and throw a more generic error